libs/designsystem/src/lib/components/list/directives/infinite-scroll.directive.ts

```diff
@@ -65,28 +71,38 @@ export class InfiniteScrollDirective implements AfterViewInit, OnDestroy {
      * Subscribe to {@link scroll$} observable and emit {@link scrollEnd} event
      * when element scroll position has surpassed the offset.
      */
-    this.scroll$
-      .pipe(
-        takeUntil(this.ngUnsubscribe$),
-        debounceTime(INFINITE_SCROLL_DEBOUNCE),
-        filter(() => !this.disabled),
-        filter((scroll) => {
-          return (
-            scroll.elementHeight * (1 - this.offset) >=
-            scroll.distanceToViewBottom - scroll.viewHeight
-          );
-        })
-      )
-      .subscribe(() => {
-        this.scrollEnd.emit();
-      });
 
+    if (!this.listComponent.isVirtualScrollEnabled) {
+      this.scroll$
+        .pipe(
+          takeUntil(this.ngUnsubscribe$),
+          debounceTime(INFINITE_SCROLL_DEBOUNCE),
+          filter(() => !this.disabled),
+          filter((scroll) => {
+            return (
+              scroll.elementHeight * (1 - this.offset) >=
+              scroll.distanceToViewBottom - scroll.viewHeight
+            );
+          })
+        )
+        .subscribe(() => {
+          this.scrollEnd.emit();
+        });
+    }
+
+    const timeoutTime = !this.listComponent.isVirtualScrollEnabled
+      ? INSTANT
+      : VIRTUAL_SCROLL_INITIALZATION;
```

Comment:
We need a higher timeout for virtual scroll to render

Reply:
Wouldn't it be better to replace our custom infinite scroll directive with Ionics own?

Reply:
Yes, it makes totally sense but might be for a separate PR and not necessary for PR.